power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/common_solver_functions.hpp

```diff
@@ -59,11 +59,26 @@ template <symmetry_tag sym> inline void copy_y_bus(YBus<sym> const& y_bus, Compl
 
 template <symmetry_tag sym>
 inline void calculate_source_result(IdxRange const& sources, Idx bus_number, YBus<sym> const& y_bus,
-                                    PowerFlowInput<sym> const& input, SolverOutput<sym>& output) {
+                                    PowerFlowInput<sym> const& input, SolverOutput<sym>& output,
+                                    ComplexValue<sym> const& i_load_gen_bus) {
+    ComplexValue<sym> const i_source_total =
+        conj(output.bus_injection[bus_number] / output.u[bus_number]) - i_load_gen_bus;
+    std::vector<ComplexTensor<sym>> y_ref_acc(sources.size());
+    std::vector<ComplexValue<sym>> i_norton_acc(sources.size());
     for (Idx const source : sources) {
         ComplexValue<sym> const u_ref{input.source[source]};
         ComplexTensor<sym> const y_ref = y_bus.math_model_param().source_param[source];
-        output.source[source].i = dot(y_ref, u_ref - output.u[bus_number]);
+        y_ref_acc[source] = y_ref;
+        i_norton_acc[source] = dot(y_ref, u_ref);
```

Comment:
This may not give you the correct indices

Reply:
@Jerry-Jinfeng-Guo You were right. It's been corrected.